src/compiler/utilities.ts

```diff
@@ -3974,8 +4009,8 @@ namespace ts {
      * @param identifier The escaped identifier text.
      * @returns The unescaped identifier text.
      */
-    export function unescapeIdentifier(identifier: string): string {
-        return identifier.length >= 3 && identifier.charCodeAt(0) === CharacterCodes._ && identifier.charCodeAt(1) === CharacterCodes._ && identifier.charCodeAt(2) === CharacterCodes._ ? identifier.substr(1) : identifier;
+    export function unescapeLeadingUnderscores(identifier: UnderscoreEscapedString): string {

```

Comment:
This fact that `identifier.text` is no longer a `string` is a breaking change and should have an entry [here](https://github.com/Microsoft/TypeScript/wiki/API-Breaking-Changes).